public/app/plugins/datasource/loki/querybuilder/parsing.ts

```diff
@@ -579,23 +580,28 @@ function getBinaryModifier(
   node: SyntaxNode | null
 ):
   | { isBool: true; isMatcher: false }
-  | { isBool: false; isMatcher: true; matches: string; matchType: 'ignoring' | 'on' }
+  | { isBool: boolean; isMatcher: true; matches: string; matchType: 'ignoring' | 'on' }
   | undefined {
   if (!node) {
     return undefined;
   }
-  if (node.getChild(Bool)) {
```

Comment:
This is what was causing the bug with the nested binary operation, the code seems to think if we have a boolean we can't have a nested expression, but our own docs demonstrate this functionality with the bool operator. 

I'm going to open up another issue to address that bool is always stripped out of this operation so this PR strictly deals with fixing these bugs that don't require adding UI